fe/fe-core/src/main/java/com/starrocks/catalog/MvRefreshArbiter.java

```diff
@@ -126,8 +126,9 @@ private static Optional<Boolean> needsToRefreshTable(MaterializedView mv,
             return Optional.of(false);
         } else if (baseTable.isNativeTableOrMaterializedView()) {
             OlapTable olapBaseTable = (OlapTable) baseTable;
+            boolean shouldRefreshBaseTable = mv.shouldRefreshBaseTable(baseTable.name);
             Set<String> baseUpdatedPartitionNames = mv.getUpdatedPartitionNamesOfOlapTable(olapBaseTable, isQueryRewrite);
-            if (!baseUpdatedPartitionNames.isEmpty()) {
+            if (!baseUpdatedPartitionNames.isEmpty() && shouldRefreshBaseTable) {
```

Comment:
` if (!baseUpdatedPartitionNames.isEmpty() && mv.shouldRefreshBaseTable(baseTable.name)) ` is better to avoid `shouldRefreshBaseTable` compute each time?

Reply:
Thanks, good advice.